src/qc_ncsa.aplf

```diff
@@ -163,6 +163,55 @@
 
     :EndNamespace
 
+    :Namespace Measurement
+        ⍝ Rolls over a set of probabilities. ⍵ should be a vector where each element indicates the probability of the vector to be in that state
+        ⍝ Returns the index of the roll
+        roll ← {1++/(?0)>+\⍵}
+
+        ⍝ Measures a vector space, returns the index of the state that was rolled. (the result will)
+        measure ← {roll (⍉⍵*2)}        
+
+        ⍝ Measures a register.
+        measure_reg ← {measure thread_reg ⍵}
+
+         split←{
+            switch←⍵[1]
+            index←⊃⍵[2]
+            grd←⊃⍵[3]
+            exc←⊃⍵[4]
+
+            grd←grd,(switch↑index)
+            index←switch↓index
+            exc←exc,(switch↑index)
+            index←switch↓index
+
+            (⍴index)=0:switch index grd exc
+            ∇ switch index grd exc
+        }
+
+         measure_qubit←{
+            ⍝ ⍺ is the qubit I want to measure
+            ⍝ ⍵ is my thread_reg
+            n_basis←1↑⍴⍵
+            n_qubits←2⍟n_basis
+            a←⍺+1
+
+            switch←⊃(n_basis÷(2*a))
+            index←⍳n_basis
+            _ _ grnd exc←split((switch)(index)(⍬)(⍬))
```

Comment:
Name these two in full.

Reply:
are names like now ok, or are they too long?

Reply:
Probably a bit long. What I meant was grnd -> ground, exc -> excited

Reply:
sorry about that, fixed it again!